### PR TITLE
GM: Trailblazer 2021-22 support 

### PR DESCRIPTION
**Pull request by TurboCE**
This pull request is for porting Trailblazer.

**Checklist**
- [x] added entry to CarInfo in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [x] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [x] route with openpilot: 5085c761395d1fe6|2023-04-07--18-20-06
- [ ] route with stock system:

**Question**
1. It is unclear whether the fingerprint was properly extracted.
2. Mass and wheelbase refer to the specifications, but I'm unsure about the other parameters.

### CODE REVIEW

File: selfdrive/car/tests/routes.py [hunk 1]
```diff
@@ -59,6 +59,7 @@
   CarTestRoute("f08912a233c1584f|2022-08-11--18-02-41", GM.BOLT_EUV, segment=1),
   CarTestRoute("555d4087cf86aa91|2022-12-02--12-15-07", GM.BOLT_EUV, segment=14),  # Bolt EV
   CarTestRoute("38aa7da107d5d252|2022-08-15--16-01-12", GM.SILVERADO),
+  CarTestRoute("5085c761395d1fe6|2023-04-07--10-32-23", GM.TRAILBLAZER),
```
Comment (adeebshihadeh) [inline selfdrive/car/tests/routes.py:62]: this route is missing the rlogs at the beginning of the route. can you replace this with another route that has rlogs for the whole time?

File: selfdrive/car/gm/values.py [hunk 2]
```diff
@@ -113,6 +114,7 @@ def init_make(self, CP: car.CarParams):
     GMCarInfo("GMC Sierra 1500 2020-21", "Driver Alert Package II", video_link="https://youtu.be/5HbNoBLzRwE"),
   ],
   CAR.EQUINOX: GMCarInfo("Chevrolet Equinox 2019-22"),
+  CAR.TRAILBLAZER: GMCarInfo("Chevrolet Trailblazer 2020-22", "Adaptive Cruise Control & LKAS"),
```
Comment (adeebshihadeh) [inline selfdrive/car/gm/values.py:117]: seems like LKAS is standard, so you only need to list ACC here

File: selfdrive/car/tests/routes.py (hunk 1 above)
Comment (TurboCE) [reply selfdrive/car/tests/routes.py:62]: Okay, I will find another one.

File: selfdrive/car/gm/values.py [hunk 3]
```diff
@@ -113,6 +114,7 @@ def init_make(self, CP: car.CarParams):
     GMCarInfo("GMC Sierra 1500 2020-21", "Driver Alert Package II", video_link="https://youtu.be/5HbNoBLzRwE"),
   ],
   CAR.EQUINOX: GMCarInfo("Chevrolet Equinox 2019-22"),
+  CAR.TRAILBLAZER: GMCarInfo("Chevrolet Trailblazer 2020-22", "Adaptive Cruise Control"),
```
Comment (sshane) [inline selfdrive/car/gm/values.py:117]: there is no 2020 Trailblazer, the first model year was 2021. Do you mean the 2020 Blazer?

Comment (TurboCE) [reply selfdrive/car/gm/values.py:117]: no, I mean the Trailblazer (Crossover) model.

It was first sold in Korea on January 16, 2020, so I assume it is a 2020 model year.

Although there are references to it being sold in the U.S. market as a 2021 model year in early 2020, please correct me if there is any inaccuracy.

File: selfdrive/car/gm/values.py (hunk 2 above)
Comment (TurboCE) [reply selfdrive/car/gm/values.py]: Thanks, 
I removed "LKAS"

File: selfdrive/car/tests/routes.py (hunk 1 above)
Comment (TurboCE) [reply selfdrive/car/tests/routes.py]: I updated it.

File: selfdrive/car/gm/values.py (hunk 3 above)
Comment (sshane) [reply selfdrive/car/gm/values.py]: Changed to 2021

File: selfdrive/car/gm/interface.py [hunk 4]
```diff
@@ -234,6 +234,15 @@ def _get_params(ret, candidate, fingerprint, car_fw, experimental_long):
       ret.centerToFront = ret.wheelbase * 0.4
       CarInterfaceBase.configure_torque_tune(candidate, ret.lateralTuning)
 
+    elif candidate == CAR.TRAILBLAZER:
+      ret.mass = 1345. + STD_CARGO_KG
+      ret.wheelbase = 2.640
```
Comment (sshane) [inline selfdrive/car/gm/interface.py:239]:
```suggestion
      ret.wheelbase = 2.64
```

**Comment by sshane** [comment]
@TurboCE can you get a route on the updated branch and verify that everything works (fingerprints, lateral is fine).

**Comment by TurboCE** [comment]
Sure, 
I will update it after get a route

**Comment by TurboCE** [comment]
Sure,
I will update it after get a route

**Comment by TurboCE** [comment]
@sshane 
After the update, openpilot was unable to identify the vehicle and operated in Dashcam mode.

5085c761395d1fe6|2023-04-13--00-09-24--0

**Comment by sshane** [comment]
@adeebshihadeh would be nice to work on GM FPv2 soon, this car fingerprints as a Silverado (too many FP in common), so the params won't be optimal

**Comment by sshane** [comment]
@TurboCE how was your driving experience so far on the Silverado car params?

**Comment by sshane** [comment]
@TurboCE merged your car under the Silverado platform for now since the fingerprints are so similar. I'll let you know once we figure out how to separate your car to use better car params!

**Comment by TurboCE** [comment]
>@[TurboCE](https://github.com/TurboCE) how was your driving experience so far on the Silverado car params?
Not bad. But it just seems a bit slow to respond.

> I'll let you know once we figure out how to separate your car to use better car params!
Okay, thanks!

**Comment by Verylukyguy** [comment]
There is a new user, Kev1n17 on Discord, with a 2023 Chevrolet Trailblazer and when he installed his Comma device, it fingerprinted as a "CHEVROLET SILVERADO 1500 2020", which seems to be normal for this platform.  This may be an indication of existing support for the 2023 model year.
He did have an issue with a "Cruise Fault: Restart the car to engage" but this is likely a separate issue.

Maybe the CARS.md file can be updated to show this.

**Pull request merged**